### PR TITLE
Add markdowns

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

Add markdowns

### CODE REVIEW

**Comment by SeaOfOcean** [comment]
Not sure where to put FlattenTable and Pack, currently in Utilities

File: docs/docs/APIdocs/layers/nn/Distance Functions/CosineDistance.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+```python
+module = CosineDistance()
+```
+
+CosineDistance creates a module that takes a table of two vectors (or matrices if in batch mode) as input and outputs the cosine distance between them.
+
+**Scala example:**
+```scala
+val module = CosineDistance()
+val t1 = Tensor(3).randn()
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Distance Functions/CosineDistance.md]: Use range to create a int input?

File: docs/docs/APIdocs/layers/nn/Loss Functions/MultiCriterion.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+## MultiCriterion ##
+
+**Scala:**
+```scala
+Scala code, how to new an instance
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Loss Functions/MultiCriterion.md]: create instance here

File: docs/docs/APIdocs/layers/nn/Loss Functions/MultiLabelSoftMarginCriterion.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+## MultiLabelSoftMarginCriterion ##
+
+**Scala:**
+```scala
+val module = MultiLabelSoftMarginCriterion()
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Loss Functions/MultiLabelSoftMarginCriterion.md]: add parameters

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/PReLU.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+## PReLU ##
+
+**Scala:**
+```scala
+val module = PReLU(nOutputPlane)
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/PReLU.md]: Add paramters' type for scala.

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/Threshold.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+```
+**Python:**
+```python
+module = Threshold(threshold, value, ip)
+```
+
+Thresholds each element of the input Tensor.
+Threshold is defined as:
+```
+         y =  x        if x >= threshold
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/Threshold.md]: add an {?

File: docs/docs/APIdocs/layers/optim/RMSprop.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+## RMSprop ##
+
+An implementation of RMSprop
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/optim/RMSprop.md]: Add an artical?

File: docs/docs/APIdocs/layers/nn/Utilities/Pack.md [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 **Scala:**
 ```scala
-val module = Pack(dim)
+val module = Pack(dim: Int)
```
Comment (qiuxin2012) [inline docs/docs/APIdocs/layers/nn/Utilities/Pack.md]: I think you can put this Pack to Merge Layers, which used to merge multi Tensor into one.

File: docs/docs/APIdocs/layers/nn/Containers/Container.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+## Container ##
+
+Container is an abstract AbstractModule class which
```
Comment (wzhongyuan) [inline docs/docs/APIdocs/layers/nn/Containers/Container.md]: "an abstract AbstractModule class" doesn't seem clear to users as there is no link to "AbstractModule", maybe "a subclass of abstract class AbstractModule" ?

File: docs/docs/APIdocs/layers/nn/Distance Functions/CosineDistance.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+	   2.0
+	   3.0
+	   [com.intel.analytics.bigdl.tensor.DenseTensor$mcF$sp of size 3]
+ }
+
+> output
+output: com.intel.analytics.bigdl.tensor.Tensor[Float] =
+0.9746319
+[com.intel.analytics.bigdl.tensor.DenseTensor of size 1]
+input
```
Comment (wzhongyuan) [inline docs/docs/APIdocs/layers/nn/Distance Functions/CosineDistance.md]: why "input" here ?

**Comment by wzhongyuan** [comment]
@SeaOfOcean  added some comments. please help check

**Comment by wzhongyuan** [comment]
@SeaOfOcean  Also I saw from you examples, some are written my float types and others are with Double. can we keep consistent to use the same type ?

**Comment by SeaOfOcean** [comment]
maybe we need a uniform type for all the examples? @jason-dai @wzhongyuan

**Comment by jason-dai** [comment]
Let's use float by default.

File: docs/docs/APIdocs/layers/nn/Loss Functions/MultiCriterion.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
+## MultiCriterion ##
+
+**Scala:**
+```scala
+val module = MultiCriterion()
```
Comment (wzhongyuan) [inline docs/docs/APIdocs/layers/nn/Loss Functions/MultiCriterion.md]: is it better to rename to criterion container or something else meaningful ? Actually it's not a module

File: docs/docs/APIdocs/layers/nn/Loss Functions/MultiLabelSoftMarginCriterion.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
+## MultiLabelSoftMarginCriterion ##
+
+**Scala:**
+```scala
+val module = MultiLabelSoftMarginCriterion(weights: Tensor[T] = null, sizeAverage: Boolean = true)
```
Comment (wzhongyuan) [inline docs/docs/APIdocs/layers/nn/Loss Functions/MultiLabelSoftMarginCriterion.md]: same as above

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/PReLU.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
+## PReLU ##
+
+**Scala:**
+```scala
+val module = PReLU(nOutputPlane: Int = 0)
```
Comment (wzhongyuan) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/PReLU.md]: remove type in argument list

File: docs/docs/APIdocs/layers/nn/Simple Layers/Transpose.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
+## Transpose ##
+
+**Scala:**
+```scala
+val module = Transpose(permutations: Array[(Int, Int)])
```
Comment (wzhongyuan) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Transpose.md]: remove type from argument list

File: docs/docs/APIdocs/layers/nn/Utilities/Pack.md [hunk 14]
```diff
@@ -0,0 +1,76 @@
+## Pack ##
+
+**Scala:**
+```scala
+val module = Pack(dim: Int)
```
Comment (wzhongyuan) [inline docs/docs/APIdocs/layers/nn/Utilities/Pack.md]: remove type from argument list

**Pull request merged**